spec/lib/services/lookup-spec.js

```diff
@@ -100,17 +101,16 @@ describe('Lookup Service', function () {
 
         it('should allow multple simultaneous cache checks', function () {
             expect(lookupService.checkNodeIdCache('testAddress')).to.be.null;
-            lookupService.checkNodeIdCache('testAddress').then(spy1, spy1);
-            lookupService.checkNodeIdCache('testAddress').then(spy2, spy2);
+            promise1 = lookupService.checkNodeIdCache('testAddress');
+            promise2 = lookupService.checkNodeIdCache('testAddress');
         });
 
-        it('should resolve pending cache checks once a value is assigned', function (done) {
+        it('should resolve pending cache checks once a value is assigned', function () {
             lookupService.assignNodeIdCache('testAddress', 'nodeId');
-            setTimeout(function () {
-                expect(spy1.called).to.be.ok;
-                expect(spy2.called).to.be.ok;
-                done();
-            }, 0);
+            return Promise.all([
```

Comment:
Although it could run in sequence, and it could pass. it's  not an essential for a pure unit test, the environment should be setup and cleared in setup(), teardown(). otherwise, it's not easy to read, maintain, debug and trace back.  not forcing to change them in this PR.